traitsui/qt4/image_enum_editor.py

```diff
@@ -99,7 +99,17 @@ def rebuild_editor(self):
         """Rebuilds the contents of the editor whenever the original factory
         object's **values** trait changes.
         """
-        self.control.model().reset()
+        # This forces the model to completely rebuild itself.
+        self.control.model().endResetModel()
```

Comment:
I am not sure when the rebuild_editor is called but I feel that the beginResetModel() and endResetModel() should always pair together so maybe they should appear on a try..finally in _update_values_and_rebuild_editor

Reply:
Do we need to call _update_values_and_rebuild_editor during the rebuild_editor?

Reply:
We might want to redsign the _update_values_and_rebuild_editor so that the begin..end reset are called in the rebuild_editor

Reply:
The `begin...` really needs to happen before the call to `values_changed` to avoid the Qt model being out of sync with the `Editor`.  I'm OK with the way I changed it; I'd be happier if it wasn't in a traits listener, but I can live with it.